### ISSUE TITLE
some question about Back Translation evaluation

### ISSUE THREAD

**Issue by ZJYCP** [BenSaunders27/ProgressiveTransformersSLP#25]
hi,
I see that the original slt model takes the images as input, and then use CNNs to get the image features. I wonder in this work, should I take the generated skeleton video or skeleton sequence as input data? 
Maybe could you please provide the back translation evalation code for us to study.
thanks a lot

**Comment by thiagomcoutinho** [comment]
I think the authors modified the back translation model to handle sign pose sequences as input, and then trained it again on PHOENIX14T, as mentioned in the second paragraph of section 4.2 of the paper. 

For sure it would be nice to have this trained model and the to evaluate our produced results.

**Comment by Tejaswini2612** [comment]
I have been trying to reproduce the back translation (BT) results as shown in the paper with **no success**!

Is anyone able to reproduce the results? It would be great to know what the settings were as I am not able to reproduce it!

Looking forward to some reply!

Thanks in advance!

**Comment by thiagomcoutinho** [comment]
Hi @Tejaswini2612, how are you trying to reproduce the BT results?

**Comment by Tejaswini2612** [comment]
@thiagomcoutinho I am using https://github.com/neccam/slt repo as mentioned in the paper with the provided settings.

**Comment by thiagomcoutinho** [comment]
@Tejaswini2612 you modified the network to take sign poses as inputs?

**Comment by Tejaswini2612** [comment]
Yes

**Comment by thiagomcoutinho** [comment]
Could you share your code please? I'm in this path right now... If I get some results closer to the paper I will let you know.

**Comment by Tejaswini2612** [comment]
The only change I made in my code is to dump the pose data in the same format as the one used in the repository I mentioned above.
Please feel free to reach out with any errors or issues that you may be encountering, I will be happy to take a look at it.

**Comment by ZJYCP** [comment]
> The only change I made in my code is to dump the pose data in the same format as the one used in the repository I mentioned above.
> Please feel free to reach out with any errors or issues that you may be encountering, I will be happy to take a look at it.

I just take the pose sequence as input and change the feature size to '151' in _origin slt code config_ , but the WER is 100 all the time and bleu4 is around 3...  Maybe Need to modify the network??

**Comment by ZJYCP** [comment]
> The only change I made in my code is to dump the pose data in the same format as the one used in the repository I mentioned above.
> Please feel free to reach out with any errors or issues that you may be encountering, I will be happy to take a look at it.

I just take the pose sequence as input and change the feature size to '151' in _origin slt code config_ , but the WER is 100 all the time and bleu4 is around 3...  Maybe Need to modify the network??

**Comment by thiagomcoutinho** [comment]
@ZJYCP , you have to feed the network with skeleton data as @Tejaswini2612 has mentioned and then change the feature size to 150 (in my understanding, you don't need the counter in this approach). If you just change the feature size to 151, you would be just limiting the image feature vector (the default data used in the SLT paper) to 151 dimensions, hence the poor results.  

My suggestion is to test your code with feature size equals to 1024 to see if the network learning, then make the changes to handle skeletons as input.

**Comment by thiagomcoutinho** [comment]
Hi @Tejaswini2612 , i just dumped the pose data in the same format as you indicated, but now the model results are poor. I have trained the model with the original data and it seems to work.

How was your loss using the skeletons? Did you make some additional changes that you could share?

**Comment by hacker009-sudo** [comment]
@ZJYCP @Clement25 please share the pretrained model, if you have stored it somewhere. Request you to please help it is really really urgent.
